### DISCUSSION TITLE
The service ID is invalid

### DISCUSSION THREAD

**Discussion by Volition42** [emailjs-com/help#2] [Q&A] [answered]
I am using the example that you provided [here](https://www.emailjs.com/docs/tutorial/creating-contact-form/). I have replaced the 'YOUR_PUBLIC_KEY' text with my public key (I even generated a new key to test if that was the issue). However, when I submit my form data I continuously get the error "FAILED... Object { status: 400, text: "The service ID is invalid. To find this ID, visit https://dashboard.emailjs.com/admin" }". I am using a free subscription but I have not even sent 1 successful email so I still have 200 in my monthly quota. How do I resolve this issue?

**Comment by Volition42** [comment]
I was able to resolve the issue after setting up MFA on my account.

**Comment by xr0master** [comment]
@Volition42 I'm glad that you installed MFA to protect your account.
However, this does not affect the functionality of the API for sending emails.

If you use the official EmailJS SDK, you will most likely not have issues if your Public ID was entered correctly, without extra spaces.

If you use the REST API, then probably, you did not specify the content type correctly.

**Comment by harperzhu** [comment]
Hi, I have the same problem, and I use the official email js SDK with React. I am 100% sure my public key is correct, but it keeps on showing that "The Public Key is invalid. To find this ID, visit https://dashboard.emailjs.com/admin/account"

**Comment by MarthaC444** [comment]
Has anyone found a common problem here? I'm having the same issue. Free plan. Testing on localhost. Following the React example [](https://www.emailjs.com/docs/examples/reactjs/), getting "bad request, 400, public key invalid". I was using the correct key. Refreshed the keys and put in the new one just to see if my key had gone magically stale. Same error with the new public key.

**Reply by harperzhu** [reply, marked as answer]
Hey! I was able to resolve it. If I remember correctly, it was because I stored my API keys in the .env and I used "" mark around it.

What I did was:
`REACT_APP_PUBLIC_KEY="my_public_key" `

 It should be `REACT_APP_PUBLIC_KEY=my_public_key`

The difference is the quotation mark around the value. 
Hope this helps

**Reply by MarthaC444** [reply]
Thanks! I ended up having a similar problem. This pointed me in the right direction.
